pages/login/LoginHandler.inc.php

```diff
@@ -169,12 +180,71 @@ function requestResetPassword($args, $request) {
 		$this->setupTemplate($request);
 		$templateMgr = TemplateManager::getManager($request);
 
+		// Check if reCaptcha is enabled and validate it if so
+		$captchaEnabled = Config::getVar('captcha', 'captcha_on_password_reset') && Config::getVar('captcha', 'recaptcha');
+		if ($captchaEnabled) {
+			import('lib.pkp.classes.form.Form');
+			$form = new Form('');
+			$form->setData('g-recaptcha-response', $request->getUserVar('g-recaptcha-response'));
+
+			import('lib.pkp.classes.form.validation.FormValidatorReCaptcha');
+			$captchaValidator = new FormValidatorReCaptcha(
+				$form,
+				$request->getRemoteAddr(),
+				'common.captcha.error.invalid-input-response',
+				$request->getServerHost()
+			);
+
+			if (!$captchaValidator->isValid()) {
+				$templateMgr->assign([
+					'captchaEnabled' => true,
+					'reCaptchaHtml' => '<div class="g-recaptcha" data-sitekey="' . Config::getVar('captcha', 'recaptcha_public_key') . '"></div><label for="g-recaptcha-response" style="display:none;">Recaptcha response</label>',
+					'error' => $captchaValidator->getMessage(),
+					'email' => $request->getUserVar('email'),
+				]);
+				$templateMgr->display('frontend/pages/userLostPassword.tpl');
+				return;
+			}
+		}
+
 		$email = $request->getUserVar('email');
 		$userDao = DAORegistry::getDAO('UserDAO'); /** @var UserDAO $userDao */
 		$user = $userDao->getUserByEmail($email);
 
+		$rateLimitingEnabled = true;
+		if ($rateLimitingEnabled && $user !== null) {
+			$sessionManager = SessionManager::getManager();
+			$session = $sessionManager->getUserSession();
+
+			// Store reset requests with timestamps in the session
+			$resetRequests = $session->getSessionVar('passwordResetRequests') ?: [];
+			$currentTime = time();
+
+			// Remove entries older than 1 hour (3600 seconds)
+			foreach ($resetRequests as $resetEmail => $timestamp) {
+				if ($currentTime - $timestamp > 3600) {
+					unset($resetRequests[$resetEmail]);
+				}
+			}
+
+			// Check if this email has a recent request (within the last 10 minutes)
+			if (isset($resetRequests[$email]) && ($currentTime - $resetRequests[$email] < 600)) {
```

Comment:
If the user was blocked, I think it's good to inform that a new request can be made in "X minutes"